src/Service/AliasService.php

```diff
@@ -46,29 +47,72 @@ public function __construct(LoggerInterface $logger, Client $elasticaClient, Env
         $this->elasticaService = $elasticaService;
     }
 
+    /**
+     * @return array<int, array{ add: array{alias: string, index: string}} | array{remove: array{alias: string, index: string}} >
+     */
+    public function atomicSwitch(Environment $environment, string $newIndex): array
+    {
+        $aliasName = $environment->getAlias();
+        $actions = [];
+        $actions[] = ['add' => ['alias' => $aliasName, 'index' => $newIndex]];
+
+        if ($oldIndex = $this->getEnvironmentIndex($environment)) {
+            $actions[] = ['remove' => ['alias' => $aliasName, 'index' => $oldIndex]];
+
+            if ($environment->isUpdateReferrers()) {
+                foreach ($this->getReferrers($oldIndex) as $referrerAlias) {
+                    if ($referrerAlias['name'] === $environment->getAlias()) {
```

Comment:
You have 4 levels of nesting. Might be thath you can in every cases list all referers and info the users about those refrerers that will be broken ?